src/DynamoCoreWpf/Views/Search/LibraryView.xaml.cs

```diff
@@ -148,7 +130,26 @@ private void OnTreeViewItemPreviewMouseLeftButton(object sender, MouseButtonEven
                 {
                     // If class button was clicked, then handle, otherwise leave it.
                     e.Handled = selectedClass.IsClassButton;
-                    selectedElement.BringIntoView();
+
+                    var classInfoPanel = wrapPanel.Children.Cast<FrameworkElement>().
+                        Where(child => child.DataContext is ClassInformationViewModel).FirstOrDefault();
+
+                    classInfoPanel.Loaded += (send, handler) =>
+                        {
+                            var selectedClassButton = WpfUtilities.FindUpVisualTree<Border>(selectedElement);
+                            var height = classInfoPanel.RenderSize.Height + selectedClassButton.RenderSize.Height;
+                            if (height > ScrollLibraryViewer.ActualHeight)
+                            {
+                                selectedClassButton.BringIntoView(new Rect(0, 0, 0, ScrollLibraryViewer.ActualHeight));
+                            }
+                            else
+                            {
+                                if (IsUserVisible(selectedClassButton, ScrollLibraryViewer))
```

Comment:
```csharp
// If the class button is already visible on the library, simply bring the 
// "StandardPanel" into view. Otherwise, the class button is brought into view.
if (IsUserVisible(selectedClassButton, ScrollLibraryViewer))
    // ...
```

As I write the above comment, I feel a little odd, if we call `selectedClassButton.BringIntoView`, will it just bring the class button into view and not the `StandardPanel`?